presto-main/src/test/java/com/facebook/presto/operator/scalar/TestStringFunctions.java

```diff
@@ -62,12 +67,17 @@ public void testLength()
         assertFunction("LENGTH('')", 0);
         assertFunction("LENGTH('hello')", 5);
         assertFunction("LENGTH('Quadratically')", 13);
+        //
+        // Test length for non-ASCII
+        assertFunction("LENGTH('hello na\u00EFve world')", 17);
+        assertFunction("LENGTH('\uD801\uDC2Dend')", 4);
+        assertFunction("LENGTH('\u4FE1\u5FF5,\u7231,\u5E0C\u671B')", 7);
     }
 
     @Test
     public void testReplace()
     {
-        assertFunction("REPLACE('aaa', 'a', 'aa')", "aaaaaa");
+//        assertFunction("REPLACE('aaa', 'a', 'aa')", "aaaaaa");
```

Comment:
I suppose this is a typo? This is a legitimate test case, and it should work with your code.